java-checks/src/main/java/org/sonar/java/checks/security/ControllingPermissionsCheck.java

```diff
@@ -0,0 +1,141 @@
+import org.sonar.plugins.java.api.tree.MethodInvocationTree;
+import org.sonar.plugins.java.api.tree.MethodTree;
+import org.sonar.plugins.java.api.tree.NewClassTree;
+import org.sonar.plugins.java.api.tree.TypeTree;
+
+@Rule(key = "S4834")
+public class ControllingPermissionsCheck extends BaseTreeVisitor implements JavaFileScanner {
+
+  private static final String MESSAGE = "Make sure that Permissions are controlled safely here.";
+
+  private static final List<String> ANNOTATIONS = Collections.unmodifiableList(Arrays.asList(
+    "org.springframework.security.access.prepost.PostAuthorize",
+    "org.springframework.security.access.prepost.PostFilter",
+    "org.springframework.security.access.prepost.PreAuthorize",
+    "org.springframework.security.access.prepost.PreFilter",
+    "org.springframework.security.access.annotation.Secured"));
+
+  private static final List<String> JSR_250_ANNOTATIONS = Collections.unmodifiableList(Arrays.asList(
+    "javax.annotation.security.RolesAllowed",
+    "javax.annotation.security.PermitAll",
+    "javax.annotation.security.DenyAll"));
+
+  private static final List<String> INTERFACES = Collections.unmodifiableList(Arrays.asList(
+    "org.springframework.security.access.AccessDecisionVoter",
+    "org.springframework.security.access.AccessDecisionManager",
+    "org.springframework.security.access.AfterInvocationProvider",
+    "org.springframework.security.access.PermissionEvaluator",
+    "org.springframework.security.access.expression.SecurityExpressionOperations",
+    "org.springframework.security.access.expression.method.MethodSecurityExpressionHandler",
+    "org.springframework.security.core.GrantedAuthority",
+    "org.springframework.security.acls.model.PermissionGrantingStrategy"));
+
+  private static final List<String> CLASSES = Collections.singletonList("org.springframework.security.config.annotation.method.configuration.GlobalMethodSecurityConfiguration");
+
+  private static final String MUTABLE_ACL_SERVICE = "org.springframework.security.acls.model.MutableAclService";
+  private static final MethodMatcherCollection METHOD_MATCHERS = MethodMatcherCollection.create(
+    MethodMatcher.create().typeDefinition(MUTABLE_ACL_SERVICE).name("createAcl").withAnyParameters(),
+    MethodMatcher.create().typeDefinition(MUTABLE_ACL_SERVICE).name("deleteAcl").withAnyParameters(),
+    MethodMatcher.create().typeDefinition(MUTABLE_ACL_SERVICE).name("updateAcl").withAnyParameters(),
+    MethodMatcher.create().typeDefinition("org.springframework.security.config.annotation.web.builders.HttpSecurity").name("authorizeRequests").withAnyParameters());
+
+  private JavaFileScannerContext context;
+
+  @Override
+  public void scanFile(JavaFileScannerContext context) {
+    if (context.getSemanticModel() == null) {
+      return;
+    }
+    this.context = context;
+    scan(context.getTree());
+  }
+
+  @Override
+  public void visitClass(ClassTree tree) {
+    if (!tree.symbol().isInterface()) {
```

Comment:
I did not spot a testcase where we would expect an issue but it is not raised because it is an interface (with a proper comment about this).